--- SDLActivity.java.SDL	2012-11-06 18:52:36.218747531 -0700
+++ SDLActivity.java	2012-11-06 23:35:52.898702646 -0700
@@ -24,6 +24,7 @@ import android.hardware.*;
 import android.content.*;
 
 import java.lang.*;
+import org.bitfighter.app.R;
 
 
 /**
@@ -57,9 +58,10 @@ public class SDLActivity extends Activit
     // Load the .so
     static {
         System.loadLibrary("SDL2");
-        //System.loadLibrary("SDL2_image");
-        //System.loadLibrary("SDL2_mixer");
-        //System.loadLibrary("SDL2_ttf");
+        System.loadLibrary("tomcrypt");
+        System.loadLibrary("tnl");
+        System.loadLibrary("luavec");
+//        System.loadLibrary("bitfighter");
         System.loadLibrary("main");
     }
 
