diff -r 77a84783ca2c lua/luajit/src/msvcbuild.bat
--- a/lua/luajit/src/msvcbuild.bat	Sat May 16 16:48:00 2015 -0600
+++ b/lua/luajit/src/msvcbuild.bat	Sat May 16 18:19:13 2015 -0600
@@ -70,6 +70,7 @@
 :NODEBUG
 @if "%1"=="amalg" goto :AMALGDLL
 @if "%1"=="static" goto :STATIC
+@if "%1"=="staticamalg" goto :STATICAMALG
 %LJCOMPILE% /MD /DLUA_BUILD_AS_DLL lj_*.c lib_*.c
 @if errorlevel 1 goto :BAD
 %LJLINK% /DLL /out:%LJDLLNAME% lj_*.obj lib_*.obj
@@ -86,6 +87,13 @@
 @if errorlevel 1 goto :BAD
 %LJLINK% /DLL /out:%LJDLLNAME% ljamalg.obj lj_vm.obj
 @if errorlevel 1 goto :BAD
+@goto :MTDLL
+:STATICAMALG
+%LJCOMPILE% ljamalg.c
+@if errorlevel 1 goto :BAD
+%LJLIB% /OUT:%LJLIBNAME% ljamalg.obj lj_vm.obj
+@if errorlevel 1 goto :BAD
+@goto :MTDLL
 :MTDLL
 if exist %LJDLLNAME%.manifest^
   %LJMT% -manifest %LJDLLNAME%.manifest -outputresource:%LJDLLNAME%;2
