diff -r 8e98c714bb2a src/video/quartz/SDL_QuartzEvents.m
--- a/src/video/quartz/SDL_QuartzEvents.m	Sat Feb 04 18:12:20 2012 -0500
+++ b/src/video/quartz/SDL_QuartzEvents.m	Tue Feb 07 11:54:27 2012 -0700
@@ -332,6 +332,10 @@
 
 static void QZ_DoKey (_THIS, int state, NSEvent *event) {
 
+    /* Pass key events on to the menu to handle cmd-keys */
+    if([[NSApp menu] performKeyEquivalent:event])
+        return;
+
     NSString *chars = NULL;
     unsigned int i, numChars;
     SDL_keysym key;
