diff -r d041dcb1951d -r a59b59cbe3d9 src/video/quartz/SDL_QuartzVideo.m
--- a/src/video/quartz/SDL_QuartzVideo.m	Fri Dec 30 16:28:41 2011 -0500
+++ b/src/video/quartz/SDL_QuartzVideo.m	Fri Dec 02 16:10:55 2011 -0500
@@ -1209,12 +1209,7 @@
                                     Uint32 flags)
 {
     /* Don't throw away the GL context if we can just resize the current one. */
-#if 0  /* !!! FIXME: half-finished side project. Reenable this if you ever debug the corner cases. */
     const BOOL save_gl = ( (video_set == SDL_TRUE) && ((flags & SDL_OPENGL) == (current->flags & SDL_OPENGL)) && (bpp == current->format->BitsPerPixel) );
-#else
-    const BOOL save_gl = NO;
-#endif
-
     NSOpenGLContext *glctx = gl_context;
     SDL_Surface* retval = NULL;
 
